src/python/WMCore/MicroService/MSPileup/MSPileupReport.py

```diff
@@ -0,0 +1,59 @@
+#!/usr/bin/env python
+#-*- coding: utf-8 -*-
+#pylint: disable=
+"""
+File       : MSPileupReport.py
+Author     : Valentin Kuznetsov <vkuznet AT gmail dot com>
+Description: MSPileup report module
+"""
+
+# system modules
+import time
+
+# WMCore modules
+from Utils.Timers import gmtimeSeconds, encodeTimestamp
+
+
+class MSPileupReport():
+    def __init__(self, autoExpire=3600):
+        self.docs = []
+        self.autoExpire = autoExpire
+
+    def addEntry(self, task, uuid, entry):
+        self.purgeExpired()
+        gmtime = gmtimeSeconds()
+        report = {'gmtime': gmtime, 'uuid': uuid,
+                  'timestamp': encodeTimestamp(gmtime),
+                  'entry': entry, 'task': task}
+        self.docs.append(report)
+
+    def purgeExpired(self):
+        """
+        Purse expired records from internal docs
+        """
+        gmtime = gmtimeSeconds()
+        for entry in list(self.docs):
+            if gmtime - entry['gmtime'] > self.autoExpire:
+                self.docs.remove(entry)
+
+    def getDocuments(self):
+        """
+        Return report documents
+        """
+        self.purgeExpired()
+        return self.docs
+
+    def getDictionary(self):
+        """
+        Return report documents in dicitonary form
+        """
+        self.purgeExpired()
+        rdict = {}
+        for doc in self.docs:
+            uuid = doc['uuid']
+            timestamp = doc['timestamp']
+            entry = doc['entry']
+            task = doc['task']
+            record = f"{timestamp} {task} task {uuid} {entry}"
+            rdict.setdefault(uuid, []).append(record)
```

Comment:
why? It is very concise and correct syntax? It actually serve many things:
- it create a dict entry with list
- it adds a new element to the list if it exists already

If I need to break it it would be more than 2 lines of code.

Reply:
As we discussed, this in the end is a matter of "taste" and I was suggesting to make it like:
```
rdict.setdefault(uuid, [])
rdict.append(record)
```
but it's good as is, there is nothing wrong!